### PR TITLE
Newline normalization hack for #276

### PR DESCRIPTION
**Pull request by janoc**
This is a quick fix for the windows newline issue, at least until a better solution could be found.

### CODE REVIEW

File: straight.el [hunk 1]
```diff
@@ -2453,6 +2453,11 @@ straight.el knows to regenerate the whole cache.")
 If this is unchanged between loading and saving the build cache,
 then the saving step is skipped for efficiency.")
 
+(defun straight--strip-newlines (str)
+  "Strips CR, LF, CRLF newlines from strings, useful for
+normalizing strings before comparison."
+  (replace-regexp-in-string "[?\n|?\r]" "" str t t))
+
```
Comment (raxod502) [inline straight.el]: Could you put this function in the section `Utility functions > Strings`, around line 307? I think it would make more sense there.

Comment (raxod502) [reply straight.el]: This regexp also strips question marks from strings. Perhaps you meant to use `[\n\t]` instead.

Comment (raxod502) [inline straight.el]: Instead of `t t`, I think the style `'fixedcase 'literal` is more clear, since it indicates what the purpose of the arguments is.

Comment (janoc) [reply straight.el]: Sure, moved it. 
Re regexp - *facepalm*. Yup, you are right, also the bar shouldn't be there. Got confused by the Emacs regexp syntax, I am not used to Lisp so much yet.

Comment (janoc) [reply straight.el]: Good point, fixed.

**Comment by janoc** [comment]
I have updated the PR with the fixes, as requested.

**Comment by raxod502** [comment]
This should not have been merged to master. I force-pushed and cherry-picked to develop instead.

This mistake has happened a number of times, so I have changed the default branch of this repository to develop.

**Comment by raxod502** [comment]
Anyway, thanks for your contribution!

**Comment by janoc** [comment]
Ah sorry, I didn't realize you had a develop branch. Thanks for merging it, though.

**Comment by raxod502** [comment]
Agreed. While I usually refrain from pushing things to master until they have been in develop for long enough to spot regressions, it seems like in this case the improvement in usability is worth the risk. I added the commit to master.

**Comment by bboytian** [comment]
Was this hot fix removed in future commits? I cant seem to find it in master anymore

**Comment by raxod502** [comment]
It's https://github.com/radian-software/straight.el/commit/62c9ee7583070514320edc6d78be9e7e56ef2822, which the GitHub interface indicates is included in master and by extension develop.

However, the variable is no longer needed because in https://github.com/radian-software/straight.el/commit/cb7679cee0ea05b589607de9080e9d1b144c92bb we moved from checking `(emacs-version)` to `emacs-version`, and the latter does not have any newlines in it.

**Pull request merged**